changelog/3853.trivial.rst

```diff
@@ -0,0 +1,2 @@
+Removed ``"run all (no recorded failures)"`` message printed with ``--failed-first`` and ``--last-failed`` when there are no failed tests.
```

Comment:
@dhirensr I took the liberty of updating the CHANGELOG a bit: we don't need to explicitly write about internal details like which tests were updated for example; those are not interesting for final users, which are the public the CHANGELOG is meant for. 😁